src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1784,55 +1784,70 @@ public bool IsOnTypeHeader(SyntaxNode root, int position)
             if (propertyDeclaration == null)
             {
                 return false;
             }
 
-            return IsInHeader(node, propertyDeclaration, propertyDeclaration.Identifier);
+            return IsOnHeader(position, node, node.Identifier);
         }
 
-        public bool IsInParameterHeader(SyntaxNode node)
+        public bool IsOnParameterHeader(SyntaxNode root, int position, out SyntaxNode parameter)
         {
-            var parameter = node.GetAncestorOrThis<ParameterSyntax>();
+            var node = TryGetAncestorForLocation<ParameterSyntax>(position, root);
+            parameter = node;
             if (parameter == null)
             {
                 return false;
             }
 
-            return IsInHeader(node, parameter, parameter.Identifier);
+            return IsOnHeader(position, node, node);
+        }
+
+        public bool IsOnMethodHeader(SyntaxNode root, int position, out SyntaxNode method)
+        {
+            var node = TryGetAncestorForLocation<MethodDeclarationSyntax>(position, root);
+            method = node;
+            if (method == null)
+            {
+                return false;
+            }
+
+            return IsOnHeader(position, node, node.ParameterList);
         }
 
-        public bool IsInMethodHeader(SyntaxNode node)
+        public bool IsOnLocalFunctionHeader(SyntaxNode root, int position, out SyntaxNode localFunction)
         {
-            var containingMethod = node.GetAncestorOrThis<MethodDeclarationSyntax>();
-            if (containingMethod == null)
+            var node = TryGetAncestorForLocation<LocalFunctionStatementSyntax>(position, root);
+            localFunction = node;
+            if (localFunction == null)
             {
                 return false;
             }
 
-            return IsInHeader(node, containingMethod, containingMethod.ParameterList);
+            return IsOnHeader(position, node, node.ParameterList);
         }
 
-        public bool IsInLocalFunctionHeader(SyntaxNode node)
+        public bool IsOnLocalDeclarationHeader(SyntaxNode root, int position, out SyntaxNode localDeclaration)
         {
-            var containingLocalFunction = node.GetAncestorOrThis<LocalFunctionStatementSyntax>();
-            if (containingLocalFunction == null)
+            var node = TryGetAncestorForLocation<LocalDeclarationStatementSyntax>(position, root);
+            localDeclaration = node;
+            if (localDeclaration == null)
             {
                 return false;
             }
 
-            return IsInHeader(node, containingLocalFunction, containingLocalFunction.ParameterList);
+            var initializersExpressions = node.Declaration.Variables
+                .Where(v => v.Initializer != null)
+                .Select(initializedV => initializedV.Initializer.Value).ToImmutableArray<SyntaxNode>();
+            return IsInHeader(position, node, node, initializersExpressions);
```

Comment:
ah.  this would be easier to read as: `IsInHeader(..., holes: initializersExpressions)`